packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -460,11 +507,17 @@ export class StateMachine extends StateMachineBase {
       tracingConfiguration: props.tracingEnabled ? this.buildTracingConfiguration() : undefined,
       ...definitionBody.bind(this, this.role, props, graph),
       definitionSubstitutions: props.definitionSubstitutions,
+      encryptionConfiguration: props.kmsKey? {
```

Comment:
nit: I *think* the style for ternary operators in this repo is to have spaces before and after the `?` and `:` (repeats)

(suggestion): Extract construction of `encryptionConfiguration`'s value to a local variable in the constructor.

Reply:
We're now constructing encryptionConfiguration like so:

```
encryptionConfiguration: constructEncryptionConfiguration(props.kmsKey, props.kmsDataKeyReusePeriodSeconds),
```

Validation is being performed when calling `constructEncryptionConfiguration()`